src/pages/LogInWithShortLivedAuthTokenPage.tsx

```diff
@@ -13,72 +12,52 @@ import useLocalize from '@hooks/useLocalize';
 import useTheme from '@hooks/useTheme';
 import useThemeStyles from '@hooks/useThemeStyles';
 import Navigation from '@libs/Navigation/Navigation';
+import type {PublicScreensParamList} from '@libs/Navigation/types';
 import * as Session from '@userActions/Session';
 import ONYXKEYS from '@src/ONYXKEYS';
+import SCREENS from '@src/SCREENS';
+import type {Account} from '@src/types/onyx';
 
-const propTypes = {
-    /** The parameters needed to authenticate with a short-lived token are in the URL */
-    route: PropTypes.shape({
-        /** Each parameter passed via the URL */
-        params: PropTypes.shape({
-            /** Short-lived authToken to sign in a user */
-            shortLivedAuthToken: PropTypes.string,
-
-            /** Short-lived authToken to sign in as a user, if they are coming from the old mobile app */
-            shortLivedToken: PropTypes.string,
-
-            /** The email of the transitioning user */
-            email: PropTypes.string,
-        }),
-    }).isRequired,
-
+type LogInWithShortLivedAuthTokenPageOnyxProps = {
     /** The details about the account that the user is signing in with */
-    account: PropTypes.shape({
-        /** Whether a sign is loading */
-        isLoading: PropTypes.bool,
-    }),
+    account: OnyxEntry<Account>;
 };
 
-const defaultProps = {
-    account: {
-        isLoading: false,
-    },
-};
+type LogInWithShortLivedAuthTokenPageProps = LogInWithShortLivedAuthTokenPageOnyxProps & StackScreenProps<PublicScreensParamList, typeof SCREENS.TRANSITION_BETWEEN_APPS>;
 
-function LogInWithShortLivedAuthTokenPage(props) {
+function LogInWithShortLivedAuthTokenPage({route, account}: LogInWithShortLivedAuthTokenPageProps) {
     const theme = useTheme();
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const {
+        params: {email, shortLivedAuthToken, shortLivedToken, exitTo, error},
+    } = route;
 
     useEffect(() => {
-        const email = lodashGet(props, 'route.params.email', '');
```

Comment:
I think we should create another `email` variable that default to `''` if `email` is undefined like it was done before, and then we can remove `email` from the the condition you put in the line 41. WDYT?